src/Compilers/CSharp/Portable/Parser/Lexer.cs

```diff
@@ -870,7 +872,19 @@ private void ScanSyntaxToken(ref TokenInfo info)
                         TextWindow.AdvanceChar();
                     }
 
-                    info.Text = TextWindow.GetText(intern: true);
+                    if (_errorCount++ > 100)
```

Comment:
I think constant is ok here. And it is better be a round decimal to signify its arbitrariness. :-)
Although I think 100 is kind of small. How many would you get if you try compiling mscorlib?  (500? 1000?)